### ISSUE TITLE
Problem with infinity dates in PostgreSQL

### ISSUE THREAD

**Issue by lsrzj** [mkleehammer/pyodbc#980]
### Environment
- Python: 3.8.0 - 64 bit
- pyodbc:  4.0.32
- OS: Linux
- DB: PostgreSQL
- driver:
[PostgreSQL Unicode]
Description=PostgreSQL ODBC driver (Unicode version)
Driver=psqlodbcw.so
Setup=libodbcpsqlS.so
Debug=0
CommLog=1
UsageCount=1

### Issue

When selecting a row that contains a -infinity date in one of it's columns, pyodbc cursor fetchall, fetchone or trying to iterate the cursor resultset gives a ValueError exception with message: ValueError: year -9999 is out of range.

**Comment by v-chojas** [comment]
https://docs.python.org/3/library/datetime.html#constants

It looks out of range to me too. Python itself has no concept of a negative infinity date. I suppose you could try to cast it to a string and retrieve it that way.

**Comment by lsrzj** [comment]
Yes, I could, but the application I'm writing generates queries dynamically and I can't foresee what is coming to make this kind of transformation during query formation. The aim of my application is to compare data between two databases and check if tables on the origin and target of the comparison has the same structure and equivalent data. By equivalent I'm saying something like, a data that came from an Oracle database, and is being stored for analytics and historical reasons on a PostgreSQL, sometimes, must be transformed on a standardized manner during ETL, due to data type incompatibilities between them. What I want to check is if the ETL process is maintaining the data integrity, that is, all fields of a given record must be equivalent. So I have to query source and target databases table by table and row by row to check it. That being said, I don't know in advance the row names and types to form the query, so, what I must do is make a simple select * from table where pkfield1 = value1 and pkfield2 = value2 on the comparison source and target databases, put both results in a dictionary and check if source == target. To make this possible, I load both objects knowing the transformation rules applied on the ETL process to transform both, source and target objects, in a common object to make the equivalence comparison possible. So everything in my application is dynamic, so I don't know how I could solve this problem of infinity dates. I was using psycopg2 for PostgreSQL, and psycopg2 deals with infinity dates without problems. For Oracle I was using cx_Oracle, but as cx_Oracle performs very, very poorly, I moved the oracle part to ODBC to retrieve the data, as I saw that PostgreSQL had an ODBC driver too, I tried to use pyodbc for PostgreSQL too to simplify things in my application, but I stumbled across the infinity date problem and returned to psycopg2.
